src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,30 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
+    m_pContext->getSingletonMap()->defineSingleton(child_group->objectName(),
+                                                   child_group);
+    child_group->hide();
+}
+
+QWidget* LegacySkinParser::parseSingletonContainer(QDomElement node) {
```

Comment:
Why does SingletonMap instantiate SingletonContainer? If it didn't then you could get rid of this method entirely and use parseStandardWidget. (also the widget's setup() method is never called which is conventional -- I could see why that would be intentional because you don't want the skinner to be able to specify a layout but OTOH you might want to specify a size constraint or BackPath for a SingletonContainer).